src/internal/http/listener/httpslistener_test.go

```diff
@@ -0,0 +1,229 @@
+	for message := range claim.Messages() {
+		session.MarkMessage(message, "")
+		//session.Commit()
+	}
+	return nil
+}
+
+type responseData struct {
+	Topic     string   `json:"topic"`
+	Consumers []string `json:"consumers"`
+}
+
+func must[R interface{}](result R, err error) R {
+	if err != nil {
+		panic(err)
+	}
+	return result
+}
+
+func runConsumer(ctx context.Context, group sarama.ConsumerGroup, topic string) {
+	go func() {
+		if err := group.Consume(ctx, []string{topic}, &mockConsumerGroupHandler{}); err != nil {
+			panic(err)
+		}
+	}()
+}
+
+func TestConsumerCount(t *testing.T) {
+	/*
+	** SETUP
+	 */
+	config := configuration.LoadInto("../../../", &configuration.Configuration{})
+
+	kafkaConfig := sarama.NewConfig()
+	kafkaConfig.Producer.Return.Successes = true
+	kafkaConfig.Consumer.Offsets.Initial = sarama.OffsetOldest
+	prodclient := must(sarama.NewClient([]string{config.KafkaClusterProdBroker}, kafkaConfig))
+	prodadmin := must(sarama.NewClusterAdminFromClient(prodclient))
+	defer prodadmin.Close()
+	devclient := must(sarama.NewClient([]string{config.KafkaClusterDevBroker}, kafkaConfig))
+	devadmin := must(sarama.NewClusterAdminFromClient(devclient))
+	defer devadmin.Close()
+
+	// setup topics
+	prodadmin.CreateTopic(topic1, &sarama.TopicDetail{
+		NumPartitions:     1,
+		ReplicationFactor: 1,
+	}, false)
+
+	prodadmin.CreateTopic(topic2, &sarama.TopicDetail{
+		NumPartitions:     1,
+		ReplicationFactor: 1,
+	}, false)
+	devadmin.CreateTopic(topic2, &sarama.TopicDetail{
+		NumPartitions:     1,
+		ReplicationFactor: 1,
+	}, false)
+
+	//wg := &sync.WaitGroup{}
+	ctx := context.Background()
+
+	// setup consumers groups and offsets
+	cg1 := must(sarama.NewConsumerGroupFromClient(consumer1, prodclient))
+	defer cg1.Close()
+	runConsumer(ctx, cg1, topic1)
+
+	cg2 := must(sarama.NewConsumerGroupFromClient(consumer2, prodclient))
+	defer cg2.Close()
+	runConsumer(ctx, cg2, topic1)
+
+	cg3 := must(sarama.NewConsumerGroupFromClient(consumer3, prodclient))
+	defer cg3.Close()
+	runConsumer(ctx, cg3, topic2)
+
+	cg4 := must(sarama.NewConsumerGroupFromClient(consumer4, devclient))
+	defer cg4.Close()
+	runConsumer(ctx, cg4, topic2)
+
+	// setup producers and generate events on topics
+	prodproducer := must(sarama.NewSyncProducerFromClient(prodclient))
+	defer prodproducer.Close()
+	devproducer := must(sarama.NewSyncProducerFromClient(devclient))
+	defer devproducer.Close()
+	_, _, err := prodproducer.SendMessage(&sarama.ProducerMessage{
+		Topic: topic1,
+		Value: sarama.StringEncoder("hello world"),
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, _, err = prodproducer.SendMessage(&sarama.ProducerMessage{
+		Topic: topic2,
+		Value: sarama.StringEncoder("hello person"),
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, _, err = devproducer.SendMessage(&sarama.ProducerMessage{
+		Topic: topic2,
+		Value: sarama.StringEncoder("hello development person"),
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
```

Comment:
Looks like SendMessage is blocking, maybe we should put these statements in a go routine?

![image](https://github.com/dfds/confluent-gateway/assets/5738476/0eb0a990-f783-4848-994f-93d12d013327)

Reply:
Blocking code in a test setup? Sign me up!

If only the consumers worked similarly we could get rid of the sleep.

I see what you are saying, that we could probably shave of some fraction of a second by waitgrouping all the producers and running them in go-subroutines.
As long as we have the 120s sleep requirement, I don't think the added complexity of asynchronous test setup is worth it. Let's dicuss :)